conans/client/tools/files.py

```diff
@@ -369,6 +370,33 @@ def dos2unix(filepath):
     _replace_with_separator(filepath, "\n")
 
 
+def rename(src, dst):
+    """
+    rename a file or folder to avoid "Access is denied" error on Windows
+    :param src: Source file or folder
+    :param dst: Destination file or folder
+    """
+    if os.path.exists(dst):
+            raise ConanException("rename {} to {} failed, dst exists.".format(src, dst))
+
+    if platform.system() == "Windows" and which("robocopy") and os.path.isdir(src):
+        # /move Moves files and directories, and deletes them from the source after they are copied.
+        # /e Copies subdirectories. Note that this option includes empty directories.
+        # /ndl Specifies that directory names are not to be logged.
+        # /nfl Specifies that file names are not to be logged.
+        completed = subprocess.run(["robocopy", "/move", "/e", "/ndl", "/nfl", src, dst],
+                                   stdout=subprocess.PIPE)
+        if completed.returncode != 1:
+            raise ConanException("rename {} to {} failed.".format(src, dst))
+    else:
+        try:
+            os.rename(src, dst)
+        except Exception as err:
+            raise ConanException("rename {} to {} failed: {}".format(src, dst, err))
+        except:
+            raise ConanException("rename {} to {} failed, unexpected error.".format(src, dst))
```

Comment:
IMO, this last catch 'em all is not needed. I would expect any error to inherit from `Exception`, have you read something different somewhere for the `os.rename` function?

```suggestion
```

Reply:
catch-all exception handling is removed.